.github/workflows/turing.yaml

```diff
@@ -354,175 +352,137 @@ jobs:
 
-          kubectl apply -f infra/e2e/vault.ingress.yaml
+          echo "::group::logs deployment/turing-mlp"
+          kubectl logs deployment/turing-mlp
+          echo "::endgroup::"
 
-      - name: Prepare cluster credentials
-        if: ${{ !matrix.useInClusterConfig }}
-        env:
-          VAULT_TOKEN: root
-        run: |
-          echo "::group::Wait for Vault to become available"
-          timeout --foreground 45 bash -c \
-            'until curl -s --fail -H "X-Vault-Token: ${{ env.VAULT_TOKEN }}" http://vault.127.0.0.1.nip.io/v1/sys/mounts; do sleep 2; done'
+          echo "::group::logs deployment/turing-merlin"
+          kubectl logs deployment/turing-merlin
           echo "::endgroup::"
 
-          tee credentials.json <<EOF
-          {
-            "master_ip": "kubernetes:443",
-            "certs": "$(k3d kubeconfig get ${{ env.CLUSTER_NAME }} | yq e '.clusters[0].cluster.certificate-authority-data' - | base64 --decode | awk 1 ORS='\\n')",
-            "client_certificate": "$(k3d kubeconfig get ${{ env.CLUSTER_NAME }} | yq e '.users[0].user.client-certificate-data' - | base64 --decode | awk 1 ORS='\\n')",
-            "client_key": "$(k3d kubeconfig get ${{ env.CLUSTER_NAME }} | yq e '.users[0].user.client-key-data' - | base64 --decode | awk 1 ORS='\\n')"
-          }
-          EOF
-
-          curl -v \
-            --header "X-Vault-Token: ${{ env.VAULT_TOKEN }}" \
-            --request POST \
-            --data @credentials.json \
-            http://vault.127.0.0.1.nip.io/v1/secret/dev
-
-          # Create a secret for Merlin with information on how to access Vault
-          kubectl create secret generic vault-secret \
-            --from-literal=vault-address=http://vault:8200 \
-            --from-literal=vault-token=${{ env.VAULT_TOKEN }}
-
-      - name: Install Turing
-        run: |
-          helm install turing ./infra/charts/turing \
-            --values infra/e2e/turing.values.yaml \
-            --values infra/e2e/${{ matrix.valuesFile }} \
-            --set turing.image.registry=${{ env.LOCAL_REGISTRY }} \
-            --set turing.image.repository=${{ github.repository }}/turing-api \
-            --set turing.image.tag=${{ env.TURING_API_VERSION }} \
-            --set turing.config.RouterDefaults.Image=${{ env.LOCAL_REGISTRY }}/${{ github.repository }}/turing-router:${{ env.TURING_ROUTER_VERSION }} \
-            --set turing.experimentEngines[0].rpcPlugin.image=${{ env.LOCAL_REGISTRY }}/${{ github.repository }}/test-experiment-engine-plugin:${{ env.TEST_EXPERIMENT_ENGINE_PLUGIN_VERSION }}
-
-      - name: Install mockserver
-        run: |
-          kubectl apply -f infra/e2e/turing.mockserver.yaml
+          echo "::group::logs deployment/turing"
+          kubectl logs deployment/turing
+          echo "::endgroup::"
+
+  test-e2e-sdk:
+    runs-on: ubuntu-latest
+    env:
+      CLUSTER_INIT_VERSION: ${{ needs.build-cluster-init.outputs.cluster-init-version }}
```

Comment:
Since these env vars are duplicated with the `test-e2e` job, shall we define them outside all jobs, at the top of the file?

Reply:
Okay I've moved these env vars out to the top of the file so they are now scoped for the entire workflow, except for those that require intermediate jobs to be completed, i.e. env vars that are defined via `${{ needs.[job_name].[var_name] }}`